dtfs-central-api/src/v1/controllers/utilisation-report-service/fee-record-correction/delete-fee-record-correction-transient-form-data.controller/index.test.ts

```diff
@@ -0,0 +1,152 @@
+    });
+
+    it('should call fee record exists once', async () => {
+      // Act
+      await deleteFeeRecordCorrectionTransientFormData(req, res);
+
+      // Assert
+      expect(mockFeeRecordExists).toHaveBeenCalledTimes(1);
+    });
+
+    it('should call fee record exists with the correct parameters', async () => {
+      // Act
+      await deleteFeeRecordCorrectionTransientFormData(req, res);
+
+      // Assert
+      expect(mockFeeRecordExists).toHaveBeenCalledWith(feeRecordId, reportId);
+    });
+
+    it('should call fee record correction transient form data delete once if a fee record exists', async () => {
+      // Arrange
+      mockFeeRecordExists.mockReturnValue(true);
+
+      // Act
+      await deleteFeeRecordCorrectionTransientFormData(req, res);
+
+      // Assert
+      expect(mockTransientFormDataDelete).toHaveBeenCalledTimes(1);
+    });
+
+    it('should call fee record correction transient form data delete with the correct parameters if a fee record exists', async () => {
+      // Arrange
+      mockFeeRecordExists.mockReturnValue(true);
+
+      // Act
+      await deleteFeeRecordCorrectionTransientFormData(req, res);
+
+      // Assert
+      expect(mockTransientFormDataDelete).toHaveBeenCalledWith(tfmUserId, feeRecordId);
+    });
+
+    it(`should respond with a '${HttpStatusCode.NotFound}' if no fee record with the provided fee record id and report id is found`, async () => {
+      // Arrange
+      mockFeeRecordExists.mockReturnValue(false);
+
+      // Act
+      await deleteFeeRecordCorrectionTransientFormData(req, res);
+
+      // Assert
+      expect(res._getStatusCode()).toEqual(HttpStatusCode.NotFound);
+      expect(mockTransientFormDataDelete).not.toHaveBeenCalled();
+    });
+
+    it("should respond with the specific error status if deleting the transient form data throws an 'ApiError'", async () => {
+      // Arrange
+      const errorStatus = HttpStatusCode.NotFound;
+      mockFeeRecordExists.mockRejectedValue(new TestApiError({ status: errorStatus }));
+
+      // Act
+      await deleteFeeRecordCorrectionTransientFormData(req, res);
+
+      // Assert
+      expect(res._getStatusCode()).toEqual(errorStatus);
+    });
+
+    it("should respond with the specific error message if deleting the transient form data throws an 'ApiError'", async () => {
+      // Arrange
+      const errorMessage = 'Some error message';
+      mockFeeRecordExists.mockRejectedValue(new TestApiError({ message: errorMessage }));
+
+      // Act
+      await deleteFeeRecordCorrectionTransientFormData(req, res);
+
+      // Assert
+      expect(res._getData()).toEqual(`Failed to delete fee record correction transient form data: ${errorMessage}`);
+    });
+
+    it(`should respond with a '${HttpStatusCode.InternalServerError}' if an unknown error occurs`, async () => {
+      // Arrange
+      mockFeeRecordExists.mockRejectedValue(new Error('Some error'));
```

Comment:
pedantic, could add a describe and add this line in a before each.  Not a blocker